src/state/translations-mixin.ts

```diff
@@ -53,10 +64,21 @@ export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
 
     protected hassConnected() {
       super.hassConnected();
+      getUserNumberFormat(this.hass!).then((format) => {
+        if (format && this.hass!.language.number_format !== format) {
+          this._selectLanguage(
+            { ...this.hass!.language, number_format: format },
+            false
+          );
+        }
+      });
       getUserLanguage(this.hass!).then((language) => {
-        if (language && this.hass!.language !== language) {
+        if (language && this.hass!.language?.language !== language) {
           // We just get language from backend, no need to save back
-          this._selectLanguage(language, false);
+          this._selectLanguage(
+            { ...this.hass!.language, language: language },
```

Comment:
We do save the `FrontendTranslationData` to the backend, but we only use the language of it and not the number format settings?

Reply:
Oh we do on the above function...

Reply:
But that means we now fetch the user data twice, we should combine the 2

Reply:
This has been combined into a single call to the backend.